Tasks/UseGoV1/installer.ts

```diff
@@ -0,0 +1,118 @@
+import * as toolLib from 'azure-pipelines-tool-lib/tool';
+import * as tl from 'azure-pipelines-task-lib/task';
+
+import * as path from 'path';
+import * as util from 'util';
+import * as os from 'os';
+
+let osPlat: string = os.platform();
+let osArch: string = os.arch();
+
+export async function getGo(version: string) {
+    // check cache
+    let toolPath: string;
+    toolPath = toolLib.findLocalTool('go', fixVersion(version));
+
+    if (!toolPath) {
+        // download, extract, cache
+        toolPath = await acquireGo(version);
+        tl.debug("Go tool is cached under " + toolPath);
+    }
+
+    setGoEnvironmentVariables(toolPath);
```

Comment:
We acquire go and set env vars but then modify the path with bin before adding to path?

Reply:
Not sure exactly what you're asking here, but I think this behavior makes sense. There are certain env vars that need to be set for Go to work well, I think it makes sense to set those before we add Go to the path - to me, once it gets added to the path that signifies that the operation has completed successfully.